sktime/utils/_testing/scenarios.py

```diff
@@ -0,0 +1,200 @@
+            etc, where method_i is method_sequence[i],
+                and args_i is arg_sequence[i]
+        and returns results.
+
+        if method_i is __init__ (a constructor),
+        obj is changed to obj.__init__(**args_i) from the next line on
+
+        Parameters
+        ----------
+        obj : class or object with methods in method_sequence
+        method_sequence : list of str, default = arg_sequence if passed
+            if arg_sequence is also None, then default = self.default_method_sequence
+            sequence of method names to be run
+        arg_sequence : list of str, default = method_sequence if passed
+            if method_sequence is also None, then default = self.default_arg_sequence
+            sequence of keys for keyword argument dicts to be used
+            names for keys need not equal names of methods
+        return_all : bool, default = False
+            whether all or only the last result should be returned
+            if False, only the last result is returned
+            if True, list of deepcopies of intermediate results is returned
+
+        Returns
+        -------
+        results: output of the last method call, if return_all = False
+            list of deepcopies of all outputs, if return_all = True
+        """
+        # if both None, fill with defaults if exist
+        if method_sequence is None and arg_sequence is None:
+            method_sequence = getattr(self, "default_method_sequence", None)
+            arg_sequence = getattr(self, "default_arg_sequence", None)
+
+        # if only one is None, fill one with the other
+        # this will also raise an error if both are still None
+        if method_sequence is None:
+            method_sequence = _check_list_of_str(arg_sequence)
+        else:
+            method_sequence = _check_list_of_str(method_sequence)
+        if arg_sequence is None:
+            arg_sequence = _check_list_of_str(method_sequence)
+        else:
+            arg_sequence = _check_list_of_str(arg_sequence)
+
+        # check that length of sequences is the same
+        num_calls = len(arg_sequence)
+        if not num_calls == len(method_sequence):
+            raise ValueError("arg_sequence and method_sequence must have same length")
+
+        # execute the commands in sequence, report result(s)
+        results = []
+        for i in range(num_calls):
+            methodname = method_sequence[i]
+            args = self.args[arg_sequence[i]]
+            if methodname != "__init__":
+                res = getattr(obj, methodname)(**args)
+            # if constructor is called, run directly and replace obj
+            else:
+                if isclass(obj):
+                    res = obj(**args)
+                else:
+                    res = type(obj)(**args)
+                obj = res
+            if return_all:
+                results += [deepcopy(res)]
+            else:
+                results = res
```

Comment:
* adding some white space to separate the two if blocks

```suggestion
            args = self.args[arg_sequence[i]]

            if methodname != "__init__":
                res = getattr(obj, methodname)(**args)
            # if constructor is called, run directly and replace obj
            else:
                if isclass(obj):
                    res = obj(**args)
                else:
                    res = type(obj)(**args)
                obj = res

            if return_all:
                results += [deepcopy(res)]
            else:
                results = res
```